### PR TITLE
feat: tls and authorization on SPU mirroring

### PR DESCRIPTION
**Pull request by fraidev**
Needs:
- [x] https://github.com/infinyon/fluvio/pull/4021

### CODE REVIEW

File: crates/fluvio-sc/src/services/auth/basic.rs [hunk 1]
```diff
@@ -91,6 +91,58 @@ mod policy {
...
+            let action = serde_json::from_str(format!("\"{}\"", action_str).as_str())
+                .map_err(Error::custom)?;
+
+            let instances = if parts.len() > 1 {
+                Some(parts[1..].iter().map(|s| s.to_string()).collect())
+            } else {
+                None
+            };
+
+            Ok(Self { action, instances })
```
Comment (sehz) [inline crates/fluvio-sc/src/services/auth/basic.rs:142]: please add test for serialize and deserialize

**Review by sehz — CHANGES_REQUESTED** [review]

File: crates/fluvio-sc/src/services/auth/basic.rs (hunk 1 above)
Comment (fraidev) [reply crates/fluvio-sc/src/services/auth/basic.rs:142]: The `test_policy_serialization` are serializing and deserializing it, does not cover it? 

Actually, this change was made first on that PR: https://github.com/infinyon/fluvio/pull/4021/files

Comment (sehz) [reply crates/fluvio-sc/src/services/auth/basic.rs:142]: Hmm.  I didn't see updated file or new file with changes

Comment (fraidev) [reply crates/fluvio-sc/src/services/auth/basic.rs:142]: https://github.com/infinyon/fluvio/pull/4021/files#diff-3542a6ea8dd07337a7a2bce491d21abb9c32597701d4ff6f3936e73293ead019R338

I added `test_action_urn_serialization` and `test_action_urn_deserialization` to test directly now anyway.

File: crates/fluvio-sc/src/services/auth/basic.rs [hunk 2]
```diff
@@ -285,29 +285,19 @@ mod test {
...
-        assert_eq!(serialized, r#""Read:user1:user2""#);
+        assert_eq!(serialized, r#""Read:user1""#);
     }
 
     #[test]
     fn test_action_urn_deserialization() {
-        let deserialized: ActionUrn = serde_json::from_str(r#""Read:user1:user2:user3""#)
-            .expect("failed to deserialize action urn");
+        let deserialized: ActionUrn =
+            serde_json::from_str(r#""Read:user1""#).expect("failed to deserialize action urn");
```
Comment (sehz) [inline crates/fluvio-sc/src/services/auth/basic.rs:297]: is this problem with rebase? why changes

Comment (fraidev) [reply crates/fluvio-sc/src/services/auth/basic.rs:297]: Oh, I think that I lost the last commit fix it after rebase, weird.

A fast fix here: https://github.com/infinyon/fluvio/pull/4024

Comment (fraidev) [reply crates/fluvio-sc/src/services/auth/basic.rs]: now it's fixed.

File: crates/fluvio-spu/src/services/public/tests/produce.rs [hunk 3]
```diff
@@ -425,7 +442,9 @@ async fn test_produce_metrics() {
     spu_config.log.base_dir = test_path;
     let ctx = GlobalContext::new_shared_context(spu_config);
 
-    let server_end_event = create_public_server(addr.to_owned(), ctx.clone()).run();
+    let auth_global_ctx =
+        SpuAuthGlobalContext::new(ctx.clone(), Arc::new(RootAuthorization::new()));
```
Comment (sehz) [inline crates/fluvio-spu/src/services/public/tests/produce.rs:446]: this is repeated. can we try to create util so all tests can just same pattern?

Comment (fraidev) [reply crates/fluvio-spu/src/services/public/tests/produce.rs]: Sure! I created a `create_public_server_with_root_auth` function for most test cases.

**Review by sehz — APPROVED** [review]
LGTM.

**Pull request merged**